### DISCUSSION TITLE
Revolt API: How to upload user avatar

### DISCUSSION THREAD

**Discussion by DezyNation** [stoatchat/discussions#633] [Q&A]
The official API documentation has mentioned a `PATCH` request to update the user. But I don't know how to upload the user's avatar.
The documentation asks for an attachment ID in the `avatar` field but how do we generate that attachment ID? Is there any API available to upload files through autumn?

**Comment by account0123** [comment]
You can generate any attachment ID if you upload a formdata to autumn. Example using Node.js:
```js
const AUTUMN = 'https://autumn.revolt.chat'
async function uploadFile(tag, buf){
    const formData = new FormData();
    formData.set("file", new Blob([buf]));
    const res = await fetch(`${AUTUMN}/${tag}`, {
        method: 'POST',
        body: formData,
    }).then(res=>res.json());
    return res.id
}
const file = fs.readFileSync('bun.gif');
uploadFile('avatars', file)
    .then(id=>console.log(`Attachment ID is ${id}`))
    .catch(e=>console.error(e));
```
Edit: ID is generated but GET requests using those IDs seem to fail :thinking:
